tests/test_app.py

```diff
@@ -1,94 +1,152 @@
 import hashlib
+from datetime import datetime, timedelta, timezone
+from urllib.parse import urljoin
 
-import pytest
-from fastapi import HTTPException
+import itsdangerous
 from fastapi.testclient import TestClient
 
-from hatch import app, config, models
+from hatch import app, config, schema, signing
 
 
 client = TestClient(app.app)
 
 
-def test_validate_raises_with_bad_secret():
-    with pytest.raises(HTTPException):
-        app.validate("bad")
+def auth_token(path, user="user", expiry=None, scope="view"):
+    if expiry is None:  # pragma: no cover
+        expiry = datetime.utcnow() + timedelta(hours=1)
```

Comment:
How come you're not testing this branch?

Reply:
Because I don't actually set a bad expiry in these tests, but I may want to in future.

I test the expiry logic in tests/test_signing, I don't repeat that test in the actual API code.

Reply:
Also this is one of those weird coverage cases - it's complaining that I've not tested the branch when `expire is not None`, basically, for which there no actual code change